spec/support/corpus.rb

```diff
@@ -124,6 +116,40 @@ def checkout
 
     private
 
+      MissingExpectedException = Class.new(StandardError)
+
+      # Count mutations and check error results against whitelist
+      #
+      # @param path [Pathname] path responsible for exception
+      #
+      # @return [Fixnum] mutations generated
+      def count_mutations_and_check_errors(path)
+        relative_path = path.relative_path_from(repo_path)
+
+        count = count_mutations(path)
+
+        expected_errors.assert_success(relative_path)
```

Comment:
> assert_success is only run if an exception isn't raised.

Ahh ok, it wasn't clear to me that `#count_mutations` could raise an exception. Maybe it should be documented using `@raise` ?